apps/cowswap-frontend/src/modules/twap/services/cancelTwapOrderTxs.ts

```diff
@@ -30,12 +44,74 @@ export function cancelTwapOrderTxs(context: CancelTwapOrderContext): MetaTransac
 }
 
 export async function estimateCancelTwapOrderTxs(context: CancelTwapOrderContext): Promise<BigNumber> {
+  if (context.chainId === SupportedChainId.LENS) {
+    // Lens is a zkSync chain, so we need to use a different estimation method
+    // See the function estimateZkSyncCancelTwapOrderTxs for details
+    return estimateZkSyncCancelTwapOrderTxs(context)
+  }
+
   const { composableCowContract, settlementContract, orderId, partOrderId } = context
-  const canelComposableCowTxCost = await composableCowContract.estimateGas.remove(orderId)
+  const cancelComposableCowTxCost = await composableCowContract.estimateGas.remove(orderId)
 
-  if (!partOrderId) return canelComposableCowTxCost
+  if (!partOrderId) return cancelComposableCowTxCost
 
   const cancelPartOrderTx = await settlementContract.estimateGas.invalidateOrder(partOrderId)
 
-  return canelComposableCowTxCost.add(cancelPartOrderTx)
+  return cancelComposableCowTxCost.add(cancelPartOrderTx)
+}
+
+/**
+ * Estimates the gas cost for cancelling a TWAP order on zkSync.
+ *
+ * Due to the fact that zkSync chains have a particular way of handling accounts, the eth_estimateGas call fails when coming from a contract.
+ * For more details, see https://github.com/zkSync-Community-Hub/zksync-developers/discussions/144
+ * For the inspiration for the workaround, see https://github.com/safe-global/safe-wallet-monorepo/pull/3369
+ *
+ * Thus, this function estimates the gas cost by simulating the transaction as if it were sent by a fake EOA address.
+ */
+async function estimateZkSyncCancelTwapOrderTxs(context: CancelTwapOrderContext): Promise<BigNumber> {
+  const { composableCowContract, settlementContract, partOrderId: hasPartOrder, chainId } = context
+
+  const orderId = getFakeTwapOrderId(chainId)
+  const cancelComposableCowTxCost = await composableCowContract.connect(FAKE_OWNER).estimateGas.remove(orderId)
+
+  if (!hasPartOrder) {
+    return cancelComposableCowTxCost
+  }
+
+  const partOrderId = await getFakeTwapPartOrderId(chainId)
```

Comment:
Can we just hardcode `partOrderId` instead of generating it?
For gas estimation it doesn't really matter

Reply:
Working on that.
But it needs to be valid orderId otherwise the estimation reverts, no?

Reply:
https://github.com/cowprotocol/cowswap/pull/6170/commits/c97401bc62b731c5d4a6ac9bbaba48124a8f1504